projects/components/src/modal/modal.ts

```diff
@@ -9,6 +9,7 @@ export interface ModalConfig<TData = unknown> {
   title?: string;
   data?: TData;
   closeOnEscapeKey?: boolean;
+  persistOnNavigate?: boolean;
```

Comment:
My original thought as well. The reason I went this direction instead is because we close on navigate in every other case. We still want that as our default behavior.

By going this direction, the original usages can be left alone and furthermore future uses can continue to omit the property to use modals in the default behavior. 

I prefer the consistency of your suggestion as well, but I think the use cases make this a more maintainable option.

Reply:
Thinking more, how would a modal stay persist when we navigate? If we navigate to a different page and if we don't destroy the modal, then that could lead to unintended side effects.

Reply:
I think that is why this property was not exposed before. cc @aaron-steinfeld if you have some thoughts on this one

Reply:
Yeah I'm not a huge fan. The use case was to change query params rather than the whole modal, and I could see supporting that (and probably by default - because it would be weird to have a query param change close a modal), but we also want to be careful about a modal changing the url in the first place.

On a slightly separate note :

> By going this direction, the original usages can be left alone and furthermore future uses can continue to omit the property to use modals in the default behavior.

If we proceed, I would still make it consistent - we can default the value to either true or false, we don't have to (and shouldn't anyway) rely on the falsiness of a missing value.